### PR TITLE
insertions to support instruction items

### PR DESCRIPTION
**Pull request by RaghavReddy**
Based on conversation to support instructions selection recipe need to support insertions.
https://docs.google.com/document/d/1KUrQrTh5M2Da4YrOVnuEHLvObEKCKJFjENzzRzW6SUg/edit

### CODE REVIEW

File: sequence_recipe.go [hunk 1]
```diff
@@ -317,6 +333,20 @@ func (cc *ContentConstraint) Validate() bool {
 	return len(cc.Errors) == 0
 }
 
+func (i *Insertion) Validate() bool {
+	if !i.Index.IsZero() {
+		if i.ContentNames == nil || len(i.ContentNames) == 0 {
```
Comment (geetavin464) [inline sequence_recipe.go]: Can we merge the 2 nested conditions into one ? like perhaps if ( !i.Index.IsZero() && ( i.contentNames == nil || len(i.contentNames) == 0 ), so it is more in alignment with the other validation

Comment (RaghavReddy) [reply sequence_recipe.go]: first condition was if Index is non-zero then we need content_names, second condition is if index is null then we don't need content_names. These checks are to make sure we need both index and content_names have to exist together.

Comment (geetavin464) [reply sequence_recipe.go]: Yes, That is right, I'm asking if we merge the nested if into 1 single if condition for readability

Comment (geetavin464) [reply sequence_recipe.go]: I mean, instead of
  if (A) 
     { if ( B || c)  }
can we just do if ( A && ( B || C) )

Comment (RaghavReddy) [reply sequence_recipe.go]: 👍 fixed it.

File: sequence_recipe.go [hunk 2]
```diff
@@ -317,6 +333,15 @@ func (cc *ContentConstraint) Validate() bool {
 	return len(cc.Errors) == 0
 }
 
+func (i *Insertion) Validate() bool {
+
+	if i.Index.IsZero() || (i.ContentNames == nil || len(i.ContentNames) == 0) {
```
Comment (geetavin464) [inline sequence_recipe.go]: Typo:   It is stated as || instead of &&    if i.Index.iszero() &&  (  .... )

Comment (RaghavReddy) [reply sequence_recipe.go]: that's correct, we need to make sure we have both index and content_names, here are the test cases for those.

https://github.com/KaplanTestPrep/atom-go/pull/6/files#diff-4ae3dc9166635806dc2a28e29ab49e8bR785

Comment (geetavin464) [reply sequence_recipe.go]: Got it. sorry, I was just looking at this block

**Pull request merged**